http/src/main/scala/quasar/destination/avalanche/http/AvalancheHttpConfig.scala

```diff
@@ -25,21 +25,29 @@ import java.net.URI
 
 import argonaut._, Argonaut._
 
+import cats.syntax.functor._
+
 import quasar.lib.jdbc.Redacted
 
 final case class AvalancheHttpConfig(
     connectionUri: URI,
     username: Username,
-    clusterPassword: ClusterPassword,
+    clusterPassword: Option[ClusterPassword],
+    googleAuth: Option[GoogleAuth],
+    salesforceAuth: Option[SalesforceAuth],
```

Comment:
I could make it an ADT, that's not a problem.

As far as external auth goes - I did use datasource-url as a starting point, but it felt weird allowing config to specify `type` and `scheme` when they _must_ be set to specific values for this to work. And then with only 2 properties I took it to what felt like its natural conclusion. I suppose my current implementation would bloat the top-level of the config if more providers are added, so it would make sense to do something more like what you pointed out to keep migrations lighter.

I'll defer to your judgment for what the right call here would be.

Reply:
I think an ADT would be preferable in the scala repr to the three `Option`s as @rintcius suggested.

Regarding the JSON encoding, I agree with @domaspoliakas that we don't need the `type` and `scheme` specified in the JSON, for the reasons already mentioned. I also agree that adding a bit of structure to make future extension less painful sounds like a good idea. The structure need not look exactly like the one from datasource-url as there we're trying to be quite general, whereas we're much more constrained here.

Reply:
Ok cool, let's drop `type` and `scheme`. 

So it looks like we all agree on a json field `"externalAuth"` which seems to map nicely to an ADT like `Auth = ClusterPassword | ExternalAuth`, right?

Reply:
I agree. The only problem is that actian may have thrown a very big spanner into my wheels for this feature. This may have to change significantly still. I'll update the PR when I know more.